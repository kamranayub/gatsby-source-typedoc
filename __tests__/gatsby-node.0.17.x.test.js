const helpers = require("./__fixtures__/helpers");
const { sourceNodes } = require("../gatsby-node");

jest.mock("typedoc", () => require("typedoc17"));

describe("gatsby-node: sourceNodes", () => {
  describe("typedoc: 0.17.x", () => {
    it("should generate project", async () => {
      const nodes = [];

      helpers.actions.createNode.mockImplementation((node) => nodes.push(node));

      await sourceNodes(helpers, {
        src: [require.resolve("./__fixtures__/simple/index.ts")],
        typedoc: {
          target: "es5",
          mode: "modules",
        },
      });

      expect(nodes).toMatchSnapshot();
    });
  });
});
