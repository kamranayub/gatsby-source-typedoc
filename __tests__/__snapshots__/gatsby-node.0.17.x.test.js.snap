// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-node: sourceNodes typedoc: 0.17.x should generate project 1`] = `
Array [
  Object {
    "id": "typedoc-default",
    "internal": Object {
      "content": "{\\"id\\":0,\\"name\\":\\"gatsby-source-typedoc\\",\\"kind\\":0,\\"flags\\":{},\\"originalName\\":\\"\\",\\"children\\":[{\\"id\\":4,\\"name\\":\\"\\\\\\"index\\\\\\"\\",\\"kind\\":1,\\"kindString\\":\\"Module\\",\\"flags\\":{\\"isExported\\":true},\\"originalName\\":\\"E:/Development/Contrib/gatsby-source-typedoc/__tests__/__fixtures__/simple/index.ts\\",\\"children\\":[{\\"id\\":5,\\"name\\":\\"TestClass1\\",\\"kind\\":128,\\"kindString\\":\\"Class\\",\\"flags\\":{\\"isExported\\":true},\\"children\\":[{\\"id\\":6,\\"name\\":\\"propPrivate\\",\\"kind\\":1024,\\"kindString\\":\\"Property\\",\\"flags\\":{\\"isPrivate\\":true,\\"isExported\\":true},\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":6,\\"character\\":21}],\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"string\\"}},{\\"id\\":7,\\"name\\":\\"propPublic\\",\\"kind\\":1024,\\"kindString\\":\\"Property\\",\\"flags\\":{\\"isPublic\\":true,\\"isExported\\":true},\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":7,\\"character\\":19}],\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"string\\"}},{\\"id\\":14,\\"name\\":\\"getterPublic\\",\\"kind\\":262144,\\"kindString\\":\\"Accessor\\",\\"flags\\":{\\"isExported\\":true},\\"getSignature\\":[{\\"id\\":15,\\"name\\":\\"__get\\",\\"kind\\":524288,\\"kindString\\":\\"Get signature\\",\\"flags\\":{\\"isExported\\":true},\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"number\\"}}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":21,\\"character\\":18}]},{\\"id\\":16,\\"name\\":\\"setterPublic\\",\\"kind\\":262144,\\"kindString\\":\\"Accessor\\",\\"flags\\":{\\"isExported\\":true},\\"setSignature\\":[{\\"id\\":17,\\"name\\":\\"__set\\",\\"kind\\":1048576,\\"kindString\\":\\"Set signature\\",\\"flags\\":{\\"isExported\\":true},\\"parameters\\":[{\\"id\\":18,\\"name\\":\\"value\\",\\"kind\\":32768,\\"kindString\\":\\"Parameter\\",\\"flags\\":{\\"isExported\\":true},\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"string\\"}}],\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"void\\"}}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":24,\\"character\\":18}]},{\\"id\\":10,\\"name\\":\\"getPrivate\\",\\"kind\\":2048,\\"kindString\\":\\"Method\\",\\"flags\\":{\\"isPrivate\\":true,\\"isExported\\":true},\\"signatures\\":[{\\"id\\":11,\\"name\\":\\"getPrivate\\",\\"kind\\":4096,\\"kindString\\":\\"Call signature\\",\\"flags\\":{\\"isExported\\":true},\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"boolean\\"}}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":13,\\"character\\":20}]},{\\"id\\":12,\\"name\\":\\"getProtected\\",\\"kind\\":2048,\\"kindString\\":\\"Method\\",\\"flags\\":{\\"isProtected\\":true,\\"isExported\\":true},\\"signatures\\":[{\\"id\\":13,\\"name\\":\\"getProtected\\",\\"kind\\":4096,\\"kindString\\":\\"Call signature\\",\\"flags\\":{\\"isExported\\":true},\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"boolean\\"}}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":17,\\"character\\":24}]},{\\"id\\":8,\\"name\\":\\"getPublic\\",\\"kind\\":2048,\\"kindString\\":\\"Method\\",\\"flags\\":{\\"isPublic\\":true,\\"isExported\\":true},\\"signatures\\":[{\\"id\\":9,\\"name\\":\\"getPublic\\",\\"kind\\":4096,\\"kindString\\":\\"Call signature\\",\\"flags\\":{\\"isExported\\":true},\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"boolean\\"}}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":9,\\"character\\":18}]}],\\"groups\\":[{\\"title\\":\\"Properties\\",\\"kind\\":1024,\\"children\\":[6,7]},{\\"title\\":\\"Accessors\\",\\"kind\\":262144,\\"children\\":[14,16]},{\\"title\\":\\"Methods\\",\\"kind\\":2048,\\"children\\":[10,12,8]}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":5,\\"character\\":23}]},{\\"id\\":19,\\"name\\":\\"exportedFn\\",\\"kind\\":64,\\"kindString\\":\\"Function\\",\\"flags\\":{\\"isExported\\":true},\\"signatures\\":[{\\"id\\":20,\\"name\\":\\"exportedFn\\",\\"kind\\":4096,\\"kindString\\":\\"Call signature\\",\\"flags\\":{\\"isExported\\":true},\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"number\\"}}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":29,\\"character\\":26}]}],\\"groups\\":[{\\"title\\":\\"Classes\\",\\"kind\\":128,\\"children\\":[5]},{\\"title\\":\\"Functions\\",\\"kind\\":64,\\"children\\":[19]}],\\"sources\\":[{\\"fileName\\":\\"index.ts\\",\\"line\\":1,\\"character\\":0}]},{\\"id\\":1,\\"name\\":\\"\\\\\\"moduleA\\\\\\"\\",\\"kind\\":1,\\"kindString\\":\\"Module\\",\\"flags\\":{\\"isExported\\":true,\\"isExternal\\":true},\\"originalName\\":\\"E:/Development/Contrib/gatsby-source-typedoc/__tests__/__fixtures__/simple/moduleA.ts\\",\\"children\\":[{\\"id\\":2,\\"name\\":\\"util\\",\\"kind\\":64,\\"kindString\\":\\"Function\\",\\"flags\\":{\\"isExported\\":true,\\"isExternal\\":true},\\"signatures\\":[{\\"id\\":3,\\"name\\":\\"util\\",\\"kind\\":4096,\\"kindString\\":\\"Call signature\\",\\"flags\\":{\\"isExported\\":true},\\"type\\":{\\"type\\":\\"intrinsic\\",\\"name\\":\\"string\\"}}],\\"sources\\":[{\\"fileName\\":\\"moduleA.ts\\",\\"line\\":1,\\"character\\":20}]}],\\"groups\\":[{\\"title\\":\\"Functions\\",\\"kind\\":64,\\"children\\":[2]}],\\"sources\\":[{\\"fileName\\":\\"moduleA.ts\\",\\"line\\":1,\\"character\\":0}]}],\\"groups\\":[{\\"title\\":\\"Modules\\",\\"kind\\":1,\\"children\\":[4,1]}]}",
      "contentDigest": "testdigest",
      "type": "Typedoc",
    },
    "source": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "flags": Object {
                    "isExported": true,
                    "isPrivate": true,
                  },
                  "id": 6,
                  "kind": 1024,
                  "kindString": "Property",
                  "name": "propPrivate",
                  "sources": Array [
                    Object {
                      "character": 21,
                      "fileName": "index.ts",
                      "line": 6,
                    },
                  ],
                  "type": Object {
                    "name": "string",
                    "type": "intrinsic",
                  },
                },
                Object {
                  "flags": Object {
                    "isExported": true,
                    "isPublic": true,
                  },
                  "id": 7,
                  "kind": 1024,
                  "kindString": "Property",
                  "name": "propPublic",
                  "sources": Array [
                    Object {
                      "character": 19,
                      "fileName": "index.ts",
                      "line": 7,
                    },
                  ],
                  "type": Object {
                    "name": "string",
                    "type": "intrinsic",
                  },
                },
                Object {
                  "flags": Object {
                    "isExported": true,
                  },
                  "getSignature": Array [
                    Object {
                      "flags": Object {
                        "isExported": true,
                      },
                      "id": 15,
                      "kind": 524288,
                      "kindString": "Get signature",
                      "name": "__get",
                      "type": Object {
                        "name": "number",
                        "type": "intrinsic",
                      },
                    },
                  ],
                  "id": 14,
                  "kind": 262144,
                  "kindString": "Accessor",
                  "name": "getterPublic",
                  "sources": Array [
                    Object {
                      "character": 18,
                      "fileName": "index.ts",
                      "line": 21,
                    },
                  ],
                },
                Object {
                  "flags": Object {
                    "isExported": true,
                  },
                  "id": 16,
                  "kind": 262144,
                  "kindString": "Accessor",
                  "name": "setterPublic",
                  "setSignature": Array [
                    Object {
                      "flags": Object {
                        "isExported": true,
                      },
                      "id": 17,
                      "kind": 1048576,
                      "kindString": "Set signature",
                      "name": "__set",
                      "parameters": Array [
                        Object {
                          "flags": Object {
                            "isExported": true,
                          },
                          "id": 18,
                          "kind": 32768,
                          "kindString": "Parameter",
                          "name": "value",
                          "type": Object {
                            "name": "string",
                            "type": "intrinsic",
                          },
                        },
                      ],
                      "type": Object {
                        "name": "void",
                        "type": "intrinsic",
                      },
                    },
                  ],
                  "sources": Array [
                    Object {
                      "character": 18,
                      "fileName": "index.ts",
                      "line": 24,
                    },
                  ],
                },
                Object {
                  "flags": Object {
                    "isExported": true,
                    "isPrivate": true,
                  },
                  "id": 10,
                  "kind": 2048,
                  "kindString": "Method",
                  "name": "getPrivate",
                  "signatures": Array [
                    Object {
                      "flags": Object {
                        "isExported": true,
                      },
                      "id": 11,
                      "kind": 4096,
                      "kindString": "Call signature",
                      "name": "getPrivate",
                      "type": Object {
                        "name": "boolean",
                        "type": "intrinsic",
                      },
                    },
                  ],
                  "sources": Array [
                    Object {
                      "character": 20,
                      "fileName": "index.ts",
                      "line": 13,
                    },
                  ],
                },
                Object {
                  "flags": Object {
                    "isExported": true,
                    "isProtected": true,
                  },
                  "id": 12,
                  "kind": 2048,
                  "kindString": "Method",
                  "name": "getProtected",
                  "signatures": Array [
                    Object {
                      "flags": Object {
                        "isExported": true,
                      },
                      "id": 13,
                      "kind": 4096,
                      "kindString": "Call signature",
                      "name": "getProtected",
                      "type": Object {
                        "name": "boolean",
                        "type": "intrinsic",
                      },
                    },
                  ],
                  "sources": Array [
                    Object {
                      "character": 24,
                      "fileName": "index.ts",
                      "line": 17,
                    },
                  ],
                },
                Object {
                  "flags": Object {
                    "isExported": true,
                    "isPublic": true,
                  },
                  "id": 8,
                  "kind": 2048,
                  "kindString": "Method",
                  "name": "getPublic",
                  "signatures": Array [
                    Object {
                      "flags": Object {
                        "isExported": true,
                      },
                      "id": 9,
                      "kind": 4096,
                      "kindString": "Call signature",
                      "name": "getPublic",
                      "type": Object {
                        "name": "boolean",
                        "type": "intrinsic",
                      },
                    },
                  ],
                  "sources": Array [
                    Object {
                      "character": 18,
                      "fileName": "index.ts",
                      "line": 9,
                    },
                  ],
                },
              ],
              "flags": Object {
                "isExported": true,
              },
              "groups": Array [
                Object {
                  "children": Array [
                    6,
                    7,
                  ],
                  "kind": 1024,
                  "title": "Properties",
                },
                Object {
                  "children": Array [
                    14,
                    16,
                  ],
                  "kind": 262144,
                  "title": "Accessors",
                },
                Object {
                  "children": Array [
                    10,
                    12,
                    8,
                  ],
                  "kind": 2048,
                  "title": "Methods",
                },
              ],
              "id": 5,
              "kind": 128,
              "kindString": "Class",
              "name": "TestClass1",
              "sources": Array [
                Object {
                  "character": 23,
                  "fileName": "index.ts",
                  "line": 5,
                },
              ],
            },
            Object {
              "flags": Object {
                "isExported": true,
              },
              "id": 19,
              "kind": 64,
              "kindString": "Function",
              "name": "exportedFn",
              "signatures": Array [
                Object {
                  "flags": Object {
                    "isExported": true,
                  },
                  "id": 20,
                  "kind": 4096,
                  "kindString": "Call signature",
                  "name": "exportedFn",
                  "type": Object {
                    "name": "number",
                    "type": "intrinsic",
                  },
                },
              ],
              "sources": Array [
                Object {
                  "character": 26,
                  "fileName": "index.ts",
                  "line": 29,
                },
              ],
            },
          ],
          "flags": Object {
            "isExported": true,
          },
          "groups": Array [
            Object {
              "children": Array [
                5,
              ],
              "kind": 128,
              "title": "Classes",
            },
            Object {
              "children": Array [
                19,
              ],
              "kind": 64,
              "title": "Functions",
            },
          ],
          "id": 4,
          "kind": 1,
          "kindString": "Module",
          "name": "\\"index\\"",
          "originalName": "E:/Development/Contrib/gatsby-source-typedoc/__tests__/__fixtures__/simple/index.ts",
          "sources": Array [
            Object {
              "character": 0,
              "fileName": "index.ts",
              "line": 1,
            },
          ],
        },
        Object {
          "children": Array [
            Object {
              "flags": Object {
                "isExported": true,
                "isExternal": true,
              },
              "id": 2,
              "kind": 64,
              "kindString": "Function",
              "name": "util",
              "signatures": Array [
                Object {
                  "flags": Object {
                    "isExported": true,
                  },
                  "id": 3,
                  "kind": 4096,
                  "kindString": "Call signature",
                  "name": "util",
                  "type": Object {
                    "name": "string",
                    "type": "intrinsic",
                  },
                },
              ],
              "sources": Array [
                Object {
                  "character": 20,
                  "fileName": "moduleA.ts",
                  "line": 1,
                },
              ],
            },
          ],
          "flags": Object {
            "isExported": true,
            "isExternal": true,
          },
          "groups": Array [
            Object {
              "children": Array [
                2,
              ],
              "kind": 64,
              "title": "Functions",
            },
          ],
          "id": 1,
          "kind": 1,
          "kindString": "Module",
          "name": "\\"moduleA\\"",
          "originalName": "E:/Development/Contrib/gatsby-source-typedoc/__tests__/__fixtures__/simple/moduleA.ts",
          "sources": Array [
            Object {
              "character": 0,
              "fileName": "moduleA.ts",
              "line": 1,
            },
          ],
        },
      ],
      "flags": Object {},
      "groups": Array [
        Object {
          "children": Array [
            4,
            1,
          ],
          "kind": 1,
          "title": "Modules",
        },
      ],
      "id": 0,
      "kind": 0,
      "kindString": undefined,
      "name": "gatsby-source-typedoc",
      "originalName": "",
    },
    "typedocId": "default",
  },
]
`;
